ament_yamllint/ament_yamllint/main.py

```diff
@@ -0,0 +1,275 @@
+            yamllint_config['rules']['line-length'] = {}
+        yamllint_config['rules']['line-length']['max'] = args.linelength
+
+    temp_config_fd, temp_config_file = tempfile.mkstemp(suffix='.yaml',
+                                                        prefix='yamllint_')
+    with os.fdopen(temp_config_fd, 'w') as f:
+        yaml.dump(yamllint_config, f)
+
+    try:
+        report = invoke_yamllint(files, config_file=temp_config_file)
+    except:  # noqa: E722
+        raise
+    finally:
+        os.remove(temp_config_file)
+
+    # generate xunit file
+    if args.xunit_file:
+        folder_name = os.path.basename(os.path.dirname(args.xunit_file))
+        file_name = os.path.basename(args.xunit_file)
+        suffix = '.xml'
+        if file_name.endswith(suffix):
+            file_name = file_name[0:-len(suffix)]
+            suffix = '.xunit'
+            if file_name.endswith(suffix):
+                file_name = file_name[0:-len(suffix)]
+        testname = '%s.%s' % (folder_name, file_name)
+
+        xml = get_xunit_content(report, testname, time.time() - start_time)
+        path = os.path.dirname(os.path.abspath(args.xunit_file))
+        if not os.path.exists(path):
+            os.makedirs(path)
+        with open(args.xunit_file, 'w') as f:
+            f.write(xml)
+
+    if any(report.values()):
+        return 1
+
+    print('No problems found, checked %d files' % (len(report),))
+    return 0
+
+
+def get_files(paths, extensions, excludes=[]):
+    files = []
+    for path in paths:
+        if os.path.isdir(path):
+            for dirpath, dirnames, filenames in os.walk(path):
+                if 'AMENT_IGNORE' in dirnames + filenames:
+                    dirnames[:] = []
+                    continue
+                # ignore folder starting with . or _
+                dirnames[:] = [d for d in dirnames if d[0] not in ['.', '_']]
+                # ignore excluded folders
+                dirnames[:] = [d for d in dirnames if d not in excludes]
+                dirnames.sort()
+
+                # select files by extension
+                for filename in sorted(filenames):
+                    if filename in excludes:
+                        continue
+                    _, ext = os.path.splitext(filename)
+                    if ext not in ['.%s' % e for e in extensions]:
+                        continue
+                    files.append(os.path.join(dirpath, filename))
+        if os.path.isfile(path):
+            files.append(path)
+    return [os.path.normpath(f) for f in files]
+
+
+def find_executable(file_name, additional_paths=None):
+    path = None
+    if additional_paths:
+        path = os.getenv('PATH', os.defpath)
+        path += os.path.pathsep + os.path.pathsep.join(additional_paths)
+    return shutil.which(file_name, path=path)
+
+
+def invoke_yamllint(files, yamllint_bin=None, config_file=None):
+    if yamllint_bin is None:
+        yamllint_bin = find_executable('yamllint')
```

Comment:
It seems unnecessary to have `yamllint_bin` be an argument here; there are no situations where we use it.   I guess I would just remove it and just unconditionally do `yamllint_bin = find_executable('yamllint')`